api_app/visualizers_manager/classes.py

```diff
@@ -288,20 +289,38 @@ def __init__(self, name: str = None):
         self._levels = {}
         self.name = name
 
-    def add_level(self, level: int, horizontal_list: VisualizableHorizontalList):
-        self._levels[level] = horizontal_list
+    def add_level(
+        self,
+        level_position: int,
+        level_size: VisualizableLevelSize = VisualizableLevelSize.S_6,
+        horizontal_list: VisualizableHorizontalList = VisualizableHorizontalList(
+            value=[]
+        ),
+    ):
+        self._levels[level_position] = {
+            "level_size": level_size,
+            "elements": horizontal_list,
+        }
+
```

Comment:
At this point i would create a `VisualizableLevel` object that in the `__init__` takes as argument
`position`, `size`, values` , the `to_dict` function will transform the object in `{"level_size": self.size, "elements": self.values.to_dict(), "level_position": self.position}` and then update the `VisualizablePage` signatures accordingly